### PR TITLE
Support for tracing the callbacks of async methods in elastiicsearch-6.x/7.x-plugin

### PR DESCRIPTION
**Pull request by CzyerChen**
### Support for tracing the callbacks of async methods in elastiicsearch-6.x/7.x-plugin
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

callbacks(onResponse/onFailure) for 
- createAsync
- deleteAsync
- analyzeAsync
- refreshAsync
- getAsync
- searchAsync
- indexAsync
- updateAsync
- scrollAsync
- searchTemplateAsync
- clearScrollAsync
- deleteByQueryAsync

image

Trace details:
ElasticsearchDeleteRequestonResponse
ElasticsearchGetReguest
ElasticsearchactionGet

record `transmission.latency` when callback:

image

7.0.0 - 7.2.1 not included in the test because analyze method appears since 7.3.0

### CODE REVIEW

File: test/plugin/scenarios/elasticsearch-7.x-scenario/support-version.list [hunk 1]
```diff
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# 7.0.0 - 7.17.12 have been tested, and 7.0.0 - 7.2.1 not included in the test
```
Comment (wu-sheng) [inline test/plugin/scenarios/elasticsearch-7.x-scenario/support-version.list:17]: It is good to see these are all verified.

File: test/plugin/scenarios/elasticsearch-7.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/elasticsearch/RestHighLevelClientCase.java [hunk 2]
```diff
@@ -97,27 +100,37 @@ public String elasticsearch() throws Exception {
...
             createIndex(indexName);
+            createIndexAsync(indexName2);
 
             // index
             index(indexName);
+            indexAsync(indexName2);
 
             // refresh
             client.indices().refresh(new RefreshRequest(indexName), RequestOptions.DEFAULT);
+            Thread.sleep(1000);
```
Comment (wu-sheng) [inline test/plugin/scenarios/elasticsearch-7.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/elasticsearch/RestHighLevelClientCase.java:111]: Do we have another way to verify the request has been verified? 
Such as, do it through query or callback?

Comment (wu-sheng) [reply test/plugin/scenarios/elasticsearch-7.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/elasticsearch/RestHighLevelClientCase.java:111]: `Sleep` usually is not stable when doing this, when resource is limited in GHA scenario.

Comment (CzyerChen) [reply test/plugin/scenarios/elasticsearch-7.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/elasticsearch/RestHighLevelClientCase.java:111]: OK, I will improve that.

**Comment by wu-sheng** [comment]
Could you verify that, is virtual database working as expected?

**Comment by CzyerChen** [comment]
> Could you verify that, is virtual database working as expected?

Looks it's working as well.
image
image
image

**Review by wu-sheng — DISMISSED** [review]

**Comment by CzyerChen** [comment]
ES-7.x case may fail again due to the limited resources.

**Comment by wu-sheng** [comment]
Test cases fail. You may need some recheck.

**Comment by CzyerChen** [comment]
> Test cases fail. You may need some recheck.

Let me think twice.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**